server/service/IndividualServicesService.js

```diff
@@ -309,29 +297,28 @@ exports.listRecords = function (user, originator, xCorrelator, traceIndicator, c
  **/
 exports.listRecordsOfApplication = function (body, user, originator, xCorrelator, traceIndicator, customerJourney) {
   return new Promise(async function (resolve, reject) {
-    let response = {};
+    let numberOfRecords = body["number-of-records"];
+    let latest = body["latest-match"];
+    let desiredApplicationName = body["application-name"];
+    let indexAlias = await getIndexAliasAsync();
     try {
-      /****************************************************************************************
-       * Setting up required local variables from the request body
-       ****************************************************************************************/
-      let applicationName = body["application-name"];
-      /****************************************************************************************
-       * Preparing response body
-       ****************************************************************************************/
-      let oamRecordProfileList = await OamRecordProfile.getRecordsListForApplicationAsync(applicationName);
-
-      /****************************************************************************************
-       * Setting 'application/json' response body
-       ****************************************************************************************/
-      response['application/json'] = oamRecordProfileList;
+      let client = await elasticsearchService.getClient(false);
+      const result = await client.search({
+        index: indexAlias,
+        from: latest,
+        size: numberOfRecords,
+        body: {
+          query: {
+            term: {
+              "application-name": desiredApplicationName
```

Comment:
This is not going to work. As the function comment here states, the field is not searchable, if it's not of type 'keyword'. I missed this while reviewing #177 .

Reply:
It is working in my case even it is not of type  keyword (it is type text )